adminapi/dataset/__init__.py

```diff
@@ -46,25 +40,23 @@ def commit(self, skip_validation=False, force_changes=False):
         elif result['status'] == 'error':
             _handle_exception(result)
 
-    def _fetch_results(self):
-        request_data = {
-            'filters': self._filters,
-            'restrict': self._restrict,
-            'order_by': self._order_by,
-        }
-        response = send_request(
-            QUERY_ENDPOINT, request_data, self.auth_token, self.timeout
-        )
-        return self._handle_response(response)
+    def get_results(self):
+        if self._results is None:
```

Comment:
Is it possible that the response is handled as None as result? If so, it can be happen, that it will try to fetch the result everytime.

Reply:
Of course, it is not possible.  They would be empty list after initialised.  That is why we check with `is None` in here.